bundles/gcp/src/main/java/org/apache/gravitino/gcs/credential/GCSTokenProvider.java

```diff
@@ -99,6 +101,57 @@ private AccessToken getToken(Set<String> readLocations, Set<String> writeLocatio
     return downscopedCredentials.refreshAccessToken();
   }
 
+  private List<String> getReadExpressions(String bucketName, String resourcePath) {
+    List<String> readExpressions = new ArrayList<>();
+    readExpressions.add(
+        String.format(
+            "resource.name.startsWith('projects/_/buckets/%s/objects/%s')",
+            bucketName, resourcePath));
+    getAllResources(resourcePath)
```

Comment:
Why do we only need to handle `readExpressions`, what about `writeExpressions`?

Reply:
It's the behaviour of Hadoop GCS connector to check subpath info to check path exists. so no need to add write permission.